pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -1264,3 +1267,34 @@ type ScalewaySDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// IonosSDConfig configurations allow retrieving scrape targets from Ionos resources.
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ionos_sd_config
+type IonosSDConfig struct {
+	// The unique ID of the Ionos data center.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	DataCenterId string `json:"datacenterID"`
+	// Port to scrape the metrics from.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+	// Refresh interval to re-read the list of resources.
+	// +optional
+	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
+	// Optional `Authorization` header configuration, required when using IONOS.
+	// Cannot be set at the same time as `basicAuth`, or `oauth2`.Optional HTTP basic authentication information.
+	// +optional
+	Authorization  *v1.SafeAuthorization `json:"authorization,omitempty"`
```

Comment:
it sounds strange to say optional and also required. I'd recommend that we make this field required since it can't reasonably work without it.
```suggestion
	// Authorization` header configuration, required when using Ionos.
	// +required
	Authorization  v1.SafeAuthorization `json:"authorization"`
```

Reply:
Yeah, that bugged me too when I checked [here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#ionos_sd_config:~:text=Optional%20HTTP%20basic%20authentication%20information%2C%20required%20when%20using%20IONOS). Will change this is next commit.

Reply:
@simonpasquier PTAL :pray: